src/server/test/db/compressedReadingsTests.js

```diff
@@ -103,39 +101,40 @@ mocha.describe('Compressed Readings', () => {
 		const endTimestamp = moment('2017-01-01').add(1, 'hour');
 		const readingValue = 10;
 		mocha.beforeEach(async () => {
+			conn = testDB.getConnection();
 			// Groups A and B will each contain a meter
 			const groupA = new Group(undefined, 'A');
 			const groupB = new Group(undefined, 'B');
 			// Group C will contain A and B
 			const groupC = new Group(undefined, 'C');
-			await Promise.all([groupA, groupB, groupC].map(group => group.insert()));
+			await Promise.all([groupA, groupB, groupC].map(group => group.insert(conn)));
 
 			// Create and get a handle to a new meter
-			await new Meter(undefined, 'Meter2', null, false, Meter.type.MAMAC).insert();
-			meter2 = await Meter.getByName('Meter2');
+			await new Meter(undefined, 'Meter2', null, false, Meter.type.MAMAC).insert(conn);
+			meter2 = await Meter.getByName('Meter2', conn);
 
 			// Place meters & groups in hierarchy
-			await groupA.adoptMeter(meter.id);
-			await groupB.adoptMeter(meter2.id);
-			await groupC.adoptGroup(groupA.id);
-			await groupC.adoptGroup(groupB.id);
-
-			// Add some readings to the meters
+			await groupA.adoptMeter(meter.id, conn);
+			await groupB.adoptMeter(meter2.id, conn);
+			await groupC.adoptGroup(groupA.id, conn);
+			await groupC.adoptGroup(groupB.id, conn);
 			const reading1 = new Reading(meter.id, readingValue, startTimestamp, endTimestamp);
```

Comment:
I think there should be a bit of spacing between adding the meters and then adding the readings. Makes for easier... reading.